pkg/build/generator/generator.go

```diff
@@ -93,12 +122,34 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
+
 	newBuild, err := g.generateBuildFromConfig(ctx, bc, request.Revision)
 	if err != nil {
 		return nil, err
 	}
+
 	glog.V(4).Infof("Build %s/%s has been generated from %s/%s BuildConfig", newBuild.Namespace, newBuild.ObjectMeta.Name, bc.Namespace, bc.ObjectMeta.Name)
-	return g.createBuild(ctx, newBuild)
+	updatedBuild, err := g.createBuild(ctx, newBuild)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(request.Image) > 0 {
+		for _, trigger := range bc.Triggers {
+			if trigger.Type != buildapi.ImageChangeBuildTriggerType {
```

Comment:
Does `ImageChangeBuildTriggerType` specify which image should trigger a change? Would you ever have multiple triggers of type ImageChangeBuildTriggerType?